ports/rp2/main.c

```diff
@@ -206,13 +206,16 @@ int main(int argc, char **argv) {
 
     soft_reset_exit:
         mp_printf(MP_PYTHON_PRINTER, "MPY: soft reboot\n");
+        #if MICROPY_PY_BLUETOOTH
+        mp_bluetooth_deinit();
+        #endif
+        #if MICROPY_PY_NETWORK_CYW43 || MICROPY_PY_BLUETOOTH_CYW43
+        cyw43_deinit(&cyw43_state);
```

Comment:
I'm not sure this is a good idea, to deinit the CYW43.  The idea is to retain the WiFi connection and IP address (at least) over a soft reset.

Reply:
Ok. I can't remember why I added it. I don't think bluetooth would need it. Maybe I thought it had been forgotten. Will remove and retest.